share/sql/upgrade-2.2-3.0.sql

```diff
@@ -20,3 +20,942 @@ CREATE TABLE monitoring.instance_availability(datetime TIMESTAMP WITH TIME ZONE
+    "expand": "'||(v_query->'expand'->>'host_id')||'",
+    "aggregate": '||q_metric_memory_agg||'
+  },
+  "metric_loadavg": {
+    "name": "metric_loadavg",
+    "record_type": "metric_loadavg_record",
+    "columns":
+    [
+      {"name": "host_id", "data_type": "INTEGER NOT NULL REFERENCES hosts (host_id)"}
+    ],
+    "history": "'||(v_query->'history'->>'host_id')||'",
+    "expand": "'||(v_query->'expand'->>'host_id')||'",
+    "aggregate": '||q_metric_loadavg_agg||'
+  },
+  "metric_vacuum_analyze": {
+    "name": "metric_vacuum_analyze",
+    "record_type": "metric_vacuum_analyze_record",
+    "columns":
+    [
+      {"name": "instance_id", "data_type": "INTEGER NOT NULL REFERENCES instances (instance_id)"},
+      {"name": "dbname", "data_type": "TEXT NOT NULL"}
+    ],
+    "history": "'||(v_query->'history'->>'dbname')||'",
+    "expand": "'||(v_query->'expand'->>'dbname')||'",
+    "aggregate": '||q_metric_vacuum_analyze_agg||'
+  },
+  "metric_replication_lag": {
+    "name": "metric_replication_lag",
+    "record_type": "metric_replication_lag_record",
+    "columns":
+    [
+      {"name": "instance_id", "data_type": "INTEGER NOT NULL REFERENCES instances (instance_id)"}
+    ],
+    "history": "'||(v_query->'history'->>'instance_id')||'",
+    "expand": "'||(v_query->'expand'->>'instance_id')||'",
+    "aggregate": '||q_metric_replication_lag_agg||'
+  },
+  "metric_replication_connection": {
+    "name": "metric_replication_connection",
+    "record_type": "metric_replication_connection_record",
+    "columns":
+    [
+      {"name": "instance_id", "data_type": "INTEGER NOT NULL REFERENCES instances (instance_id)"},
+      {"name": "upstream", "data_type": "TEXT NOT NULL"}
+    ],
+    "history": "'||(v_query->'history'->>'upstream')||'",
+    "expand": "'||(v_query->'expand'->>'upstream')||'",
+    "aggregate": '||q_metric_replication_connection_agg||'
+  },
+  "metric_heap_bloat": {
+    "name": "metric_heap_bloat",
+    "record_type": "metric_bloat_ratio_record",
+    "columns":
+    [
+      {"name": "instance_id", "data_type": "INTEGER NOT NULL REFERENCES instances (instance_id)"},
+      {"name": "dbname", "data_type": "TEXT NOT NULL"}
+    ],
+    "history": "'||(v_query->'history'->>'dbname')||'",
+    "expand": "'||(v_query->'expand'->>'dbname')||'",
+    "aggregate": '||q_metric_bloat_ratio_agg||'
+  },
+  "metric_btree_bloat": {
+    "name": "metric_btree_bloat",
+    "record_type": "metric_bloat_ratio_record",
+    "columns":
+    [
+      {"name": "instance_id", "data_type": "INTEGER NOT NULL REFERENCES instances (instance_id)"},
+      {"name": "dbname", "data_type": "TEXT NOT NULL"}
+    ],
+    "history": "'||(v_query->'history'->>'dbname')||'",
+    "expand": "'||(v_query->'expand'->>'dbname')||'",
+    "aggregate": '||q_metric_bloat_ratio_agg||'
+  }}')::JSON INTO v_conf;
+  RETURN v_conf;
+
+END;
+
+$$;
+
+SET search_path TO monitoring;
+
+SELECT * FROM monitoring.create_tables();
```

Comment:
This won't work since `create_tables` hasn't been updated.
Should we first execute `monitoring.sql` or should we update the function here?

Reply:
FYI, here's the error I got when executing `upgrade-2.2-3.0.sql`
```
psql:share/sql/upgrade-2.2-3.0.sql:936: ERROR:  type "metric_temp_files_size_tblspc_record" does not exist
LIGNE 1 : ...nces (instance_id), spcname TEXT NOT NULL, record metric_tem...
                                                               ^
REQUÊTE : CREATE TABLE metric_temp_files_size_tblspc_current (datetime TIMESTAMPTZ NOT NULL, instance_id INTEGER NOT NULL REFERENCES instances (instance_id),
 spcname TEXT NOT NULL, record metric_temp_files_size_tblspc_record)
CONTEXTE : PL/pgSQL function create_tables() line 29 at EXECUTE
```

Reply:
Fixed it. Thanks.